src/hotspot/cpu/x86/stubGenerator_x86_64_string.cpp

```diff
@@ -0,0 +1,1051 @@
+        __ movq(needle_val, Address(needle, 1));
+
+        __ align(16);
+        __ bind(L_innerLoop);
+        __ tzcntl(set_bit, eq_mask);
+        __ cmpq(Address(hs_ptr, set_bit, Address::times_1, 1), needle_val);
+        __ je(L_bigCaseFixupAndReturn);
+        __ blsrl(eq_mask, eq_mask);
+        __ jne(L_innerLoop);
+        __ jmp(L_loopTop);
+      }
+
+
+
+  // Big case 10:
+      large_hs_jmp_table[9] = __ pc();
+      {
+        Label L_loopTop, L_innerLoop;
+
+        string_indexof_big_loop_helper(10, L_checkRangeAndReturn, L_loopTop);
+        __ movq(needle_val, Address(needle, 1));
+
+        __ align(16);
+        __ bind(L_innerLoop);
+        __ tzcntl(set_bit, eq_mask);
+        __ cmpq(Address(hs_ptr, set_bit, Address::times_1, 1), needle_val);
+        __ je(L_bigCaseFixupAndReturn);
+        __ blsrl(eq_mask, eq_mask);
+        __ jne(L_innerLoop);
+        __ jmp(L_loopTop);
+      }
+    }
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+    ////////////////////////////////////////////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////////////////
+    __ align(8);
+
+    jump_table = __ pc();
+
+    for (jmp_ndx = 0; jmp_ndx < 10; jmp_ndx++) {
+      __ emit_address(large_hs_jmp_table[jmp_ndx]);
+    }
+
+    jump_table_1 = __ pc();
+
+    for (jmp_ndx = 0; jmp_ndx < 10; jmp_ndx++) {
+      __ emit_address(small_hs_jmp_table[jmp_ndx]);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////////////////
+    //
+    // Big case default:
+
+    {
+      Label L_loopTop, L_loopMid, L_innerLoop, L_found, L_bigDefaultNotFound;
+
+      __ bind(L_bigCaseDefault);
+      __ movq(r11, -1);
+      __ testq(rsi, rsi);
+      __ jle(L_checkRangeAndReturn);
+      __ movq(Address(rsp, 0x20), r10);
+      __ leaq(rax, Address(rbx, rsi, Address::times_1));
+      __ movq(Address(rsp, 0x10), rax);
+      __ vpbroadcastb(xmm0, Address(r14, 0), Assembler::AVX_256bit);
+      __ vmovdqu(Address(rsp, 0x50), xmm0);
+      __ vpbroadcastb(xmm0, Address(r12, r14, Address::times_1, -1), Assembler::AVX_256bit);
+      __ vmovdqu(Address(rsp, 0x30), xmm0);
+      __ movl(rax, rsi);
+      __ subl(rax, r12);
+      __ incl(rax);
+      __ andl(rax, 0x1f);
+      __ cmpq(rsi, 0x21);
+      __ movl(rcx, 0x20);
+      __ cmovl(Assembler::aboveEqual, rcx, rax);
+      __ incq(r14);
+      __ movq(rax, rbx);
+      __ movq(rbx, r14);
+      __ leaq(r15, Address(r12, -0x2));
```

Comment:
Kindly use semantically meaningful names instead of direct GPR names.

Reply:
Done.